Modules/Common/src/ReferenceComparatorCheck.cxx

```diff
@@ -139,17 +169,22 @@ Quality ReferenceComparatorCheck::check(std::map<std::string, std::shared_ptr<Mo
   Quality result = Quality::Good;
   for (auto& [key, mo] : *moMap) {
     auto moName = mo->getName();
+    Quality quality;
+    std::string message;
 
-    auto* canvas = dynamic_cast<TCanvas*>(mo->getObject());
-    if (!canvas) {
+    // run the comparison algorithm
+    if (mo->getObject()->IsA() == TClass::GetClass<TCanvas>()) {
+      // We got a canvas. It contains the plot and its reference.
+      auto* canvas = dynamic_cast<TCanvas*>(mo->getObject());
+      quality = compare(canvas, mComparator.get(), message);
+    } else if (mo->getObject()->InheritsFrom(TH1::Class())) {
+      // We got a plot, we have to find the reference before calling the comparator
+      quality = getSinglePlotQuality(mo, message);
+    } else {
+      ILOG(Warning, Ops) << "mo is not a TCanvas or a TH1" << ENDM;
```

Comment:
If it's a message for an operator, it could be more helpful and precise perhaps?
```suggestion
      ILOG(Warning, Ops) << "Compared Monitor Object '" << mo->getName() << "` is not a TCanvas or a TH1, the detector QC responsible should review the configuration" << ENDM;
```